Lib/test/test_functools.py

```diff
@@ -1495,6 +1495,59 @@ def square(x):
             self.assertEqual(square.cache_info().hits, 4)
             self.assertEqual(square.cache_info().misses, 4)
 
+    def test_lru_with_container_types(self):
+        def identity(x):
+            return x
+
+        values = [(), (1, 2), (1, 2, 'a')]
+        for maxsize in (None, 128):
+            for current_value in values:
+                with self.subTest(current_value=current_value, maxsize=maxsize):
+                    cached = self.module.lru_cache(maxsize, typed=True)(identity)
+
+                    # some unrelated tuple:
+                    cached((1, 2, 3, 4, 5))  # miss
+                    cached((1, 2, 3, 4, 5))  # hit
+
+                    cached(current_value)  # miss
+                    res = cached(current_value)  # hit
+
+                    self.assertEqual(res, current_value)
+                    self.assertEqual(cached.cache_info().hits, 2)
+                    self.assertEqual(cached.cache_info().misses, 2)
+
+    def test_lru_with_container_types_hash_collision(self):
+        # https://bugs.python.org/issue45701
```

Comment:
We would normally only reference a bug if it pertained to a bug in the lru_cache itself.  In this case, the referenced bug was in application code.  Linking to it from this from here doesn't add any explanatory value.